rayon-core/src/configuration.rs

```diff
@@ -140,6 +156,27 @@ impl Configuration {
         self
     }
 
+    /// Returns a copy of the current thread start callback.
+    pub fn start_handler(&self) -> Option<StartHandler> {
+        self.start_handler.clone()
+    }
+
+    /// Sets a callback to be invoked on thread start.
```

Comment:
Let's expand this doc comment. For example, I would add a paragraph about panics:

"If this closure should panic, the panic will be passed to the panic handler. If that handler returns, then startup will continue as normal."